test/point.spec.ts

```diff
@@ -0,0 +1,325 @@
+      expect(p.subPoints(a, b)).to.eql(p.point(3, 3))
+    })
+    it("can subtract one point with a number", function () {
+      const a = p.point(5, 5)
+      const b = 2
+
+      // 5 -  2 = 3
+      expect(p.subPoints(a, b)).to.eql(p.point(3, 3))
+    })
+  })
+
+  describe("mulPoints(a, b)", function() {
+    it("can multiply one point with another", function () {
+      const a = p.point(5, 5)
+      const b = p.point(2, 2)
+
+      // 5 * 2 = 10
+      expect(p.mulPoints(a, b)).to.eql(p.point(10, 10))
+    })
+    it("can multiply a point with a number", function () {
+      const a = p.point(5, 5)
+      const b = 2
+
+      // 5 * 2 = 10
+      expect(p.mulPoints(a, b)).to.eql(p.point(10, 10))
+    })
+  })
+
+  describe("divPoint(a, b)", function() {
+    it("can divide two points with each other", function () {
+    const a = p.point(10, 10)
+    const b = p.point(2, 2)
+
+      // 10 / 2 = 5
+      expect(p.divPoint(a, b)).to.eql(p.point(5, 5))
+    })
+    it("can divide one point with number", function () {
+      const a = p.point(10, 10)
+      const b = 2
+
+      // 10 / 2 = 5
+      expect(p.divPoint(a, b)).to.eql(p.point(5, 5))
+    })
+  })
+
+})
+
+
+describe("Bounds", function () {
+  describe("boundsFor(a, b, d)", function () {
+    it("creates an object with shape { ax, ay, bx, by, dx, dy }", function () {
+
+      const a = p.point(0,0)
+      const b = p.point(1,0)
+      const d = p.point(0,1)
+
+      // So from "point.ts"
+      // Bounds are represented by three points A, B and D such that:
+      //     0,0 A---B 1,0
+      //         |   |
+      //     0,1 D---C 1,1 <--- Inferred
+      // Given that bounds are always rectangles, the missing point C can be inferred.
+
+      const bounds = p.boundsFor(a, b, d)
+
+      expect(bounds).to.have.keys(["ax", "ay", "bx", "by", "dx", "dy"])
+    })
+  })
+
+  describe("bc(b)", function () {
+    it("Inferres the correct bounds for Bounds-C", function () {
+
+      const a = p.point(0,0)
+      const b = p.point(1,0)
+      const d = p.point(0,1)
+
+      const bounds = p.boundsFor(a, b, d)
+      // See above for explanation
+      const boundsC = p.bc(bounds)
+
+      expect(boundsC).to.eql(p.point(1, 1))
+    })
+  })
+
+  describe("axisAligned(b)", function () {
+    it("Correctly identifies two aligning axis", function () {
+
+      const a = p.point(0, 0)
+      const b = p.point(1, 0)
+      const d = p.point(0, 1)
+
+      const bounds = p.boundsFor(a, b, d)
+      
+      // b.ax === b.dx && b.ay === b.by
+      const isAligned = p.axisAligned(bounds)
+
+      expect(isAligned).to.be.true
+    })
+    it("Correctly identifies non-aligned bounds", function () {
+
+      const a = p.point(0, 0)
+      const b = p.point(6, 0)
+      const d = p.point(-1, 2)
+      
+
+      // This is an example of what that looks like:
+      //     0,0 A-----B 6,0
+      //        /     /
+      //       /     /
+      // -1,2 D-----C 5,2
+      // (not to scale lol)
```

Comment:
next level testing

Reply:
I mean,  didn't know how to explain what I was testing with words!